src/linker/Linker/Driver.cs

```diff
@@ -643,9 +668,13 @@ static IStep FindStep (Pipeline pipeline, string name)
 			return null;
 		}
 
-		static IStep ResolveStep (string type)
+		static IStep ResolveStep (string type, Assembly assembly)
 		{
 			Type step = Type.GetType (type, false);
+			if (step == null) {
```

Comment:
The logic has to be inverted, if assembly != null it has to loop up the type in the assembly first